tasks/eth/008-fp-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,263 @@
+        _proxies.SuperchainConfig = stdJson.readAddress(addressesJson, "$.superchain_config_addr");
+    }
+ 
+    function _postCheck(Vm.AccountAccess[] memory accesses, SimulationPayload memory simPayload)
+        internal
+        view
+        override
+    {
+        console.log("Running post-deploy assertions");
+        checkSemvers();
+        checkStateDiff(accesses);
+        checkSystemConfig();
+        checkOptimismPortal();
+        console.log("All assertions passed!");
+    }
+
+    function checkStateDiff(Vm.AccountAccess[] memory accountAccesses) internal view override {
+        require(accountAccesses.length > 0, "No account accesses");
+
+        super.checkStateDiff(accountAccesses);
+
+        // Assert that no other OP Contracts other than the SystemConfig and the OptimismPortal are accessed
+        for (uint256 i; i < accountAccesses.length; i++) {
+            Vm.AccountAccess memory accountAccess = accountAccesses[i];
+            require(
+                accountAccess.oldBalance == accountAccess.newBalance,
+                string.concat("Unexpected balance change: ", vm.toString(accountAccess.account))
+            );
+            require(
+                accountAccess.kind != VmSafe.AccountAccessKind.SelfDestruct,
+                string.concat("Self-destructed account: ", vm.toString(accountAccess.account))
+            );
+
+            for (uint256 j; j < accountAccess.storageAccesses.length; j++) {
+                Vm.StorageAccess memory storageAccess = accountAccess.storageAccesses[j];
+
+                address account = storageAccess.account;
+                require(account != proxies.L1CrossDomainMessenger, "state-000");
+                require(account != proxies.L1ERC721Bridge, "state-100");
+                require(account != proxies.L1StandardBridge, "state-200");
+                require(account != proxies.OptimismMintableERC20Factory, "state-300");
+            }
+        }
+    }
+
+    function checkSemvers() internal view {
+        // These are the expected semvers based on the `op-contracts/v1.4.0-rc.4` release.
+        // https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2Fv1.4.0-rc.4
+        require(ISemver(proxies.L1CrossDomainMessenger).version().eq("2.3.0"), "semver-100");
+        require(ISemver(proxies.L1ERC721Bridge).version().eq("2.1.0"), "semver-700");
+        require(ISemver(proxies.L1StandardBridge).version().eq("2.1.0"), "semver-200");
+        require(ISemver(proxies.OptimismMintableERC20Factory).version().eq("1.9.0"), "semver-400");
+        require(ISemver(proxies.OptimismPortal).version().eq("3.10.0"), "semver-500");
+        require(ISemver(proxies.SystemConfig).version().eq("2.2.0"), "semver-600");
+        require(ISemver(proxies.ProtocolVersions).version().eq("1.0.0"), "semver-800");
+        require(ISemver(proxies.SuperchainConfig).version().eq("1.1.0"), "semver-900");
```

Comment:
Should we check the DGF as well?

Reply:
100%. it would have caught the issue clabby noted.

Reply:
Yup. Forgot to take care of this after upgrading the optimism submodule. Fixed.